lib/rabbitmq/http/client.rb

```diff
@@ -404,14 +404,15 @@ def initialize_connection(endpoint, options = {})
         user     = uri.user     || options.delete(:username) || "guest"
         password = uri.password || options.delete(:password) || "guest"
         options = options.merge(:url => uri.to_s)
+        adapter = options.delete(:adapter) || Faraday.default_adapter
```

Comment:
And the difference being: if an option that's unknown to Faraday is not deleted from `options`, e.g. `:username`, that will result in an exception.